### ISSUE TITLE
Make the search function more accessible

### ISSUE THREAD

**Issue by simomps** [digitalfabrik/entitlementcard#2723]
### Describe the Problem

Currently you have to interact with 23 items when trying to search with a own search text
Even when you dont have any search results, you only get that information after those 23 interactions

(Additionally -> https://github.com/digitalfabrik/entitlementcard/issues/2716?issue=digitalfabrik%7Centitlementcard%7C3021 Reposition and change the search sorting button )

### Describe Your Preferred Solution
Using screen reader:
- Category list item should not be separated into Icon and Text but should be only one selectable element
- It should be possible to skip the category list without looping through every single item

### Describe Possible Alternatives

-

### Related Issues

https://github.com/digitalfabrik/entitlementcard/issues/2716

https://github.com/digitalfabrik/entitlementcard/issues/2720 Make search icon clickable

https://github.com/digitalfabrik/entitlementcard/issues/2721 Combine category-icon and category-text

https://github.com/digitalfabrik/entitlementcard/issues/2722 Improve contrast of already selected categories

https://github.com/digitalfabrik/entitlementcard/issues/2716?issue=digitalfabrik%7Centitlementcard%7C3021 Reposition and change the search sorting button

### Additional Information

<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/8d257c16-5021-4f1c-869e-871ee375f593" />
